app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -388,5 +404,44 @@ void handleDataDelivery(RightMeshEvent e) {
             unDeliveredMessageIds.remove(deliveryDataId);
             updateInterface();
         }
+        else if(undeliveredPeerUpdateMessage.containsKey(deliveryDataId)){
+            // removing the message from undeliveredPeerUpdate map since we dont need to send it again.
+            undeliveredPeerUpdateMessage.remove(deliveryDataId);
+        }
+    }
+
+    /**
+     * Runs a timer task when there are undelivered peer updates in the que
+     */
+    void setTimerRunningForUndeliveredPackages(){
+        if(!isTimerRunning && !undeliveredPeerUpdateMessage.isEmpty()){
+            isTimerRunning = true;
+            undeliveredPackageTimer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    //check if there are any undelivered packages
+                    if(!undeliveredPeerUpdateMessage.isEmpty()){
+                        byte[] message = createPeerUpdatePayloadFromUser(user);
+                        for(Map.Entry<Integer,MeshId> meshIdEntry: undeliveredPeerUpdateMessage.entrySet()){
+                            MeshId peerId = meshIdEntry.getValue();
+                            try {
+                                //send the packages again if there are undelivered packages
+                                meshManager.sendDataReliable(peerId, MESH_PORT, message);
+                            } catch (RightMeshException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                        // All undelivered messages were sent again so Map is empty now.
+                        undeliveredPeerUpdateMessage.clear();
```

Comment:
They are sent again, but aren't necessarily delivered. Should this continue running until the messages are removed by a received event?

Reply:
Feel like that could create unnecessary thread running all the time. and if a peer gets disconnected, it would keep sending data again and again.